presto-spi/src/main/java/com/facebook/presto/spi/ConnectorPageSink.java

```diff
@@ -14,10 +14,11 @@
 package com.facebook.presto.spi;
 
 import com.facebook.presto.spi.block.Block;
+import io.airlift.slice.Slice;
 
 public interface ConnectorPageSink
 {
     void appendPage(Page page, Block sampleWeightBlock);
 
-    String commit();
+    Iterable<Slice> commit();
```

Comment:
Is there any place where we actually need to return an Iterable (vs a List or Set)? It's generally better to return the most concrete type that works for the abstraction.